src/index.ts

```diff
@@ -196,16 +196,17 @@ export class Datastore {
 
     /**
      * Sets the storage and encryption settings for the Datastore
+     * @param storageProvider Storage provider
      * @param host Host
      * @param port Port 
      * @param protocol HTTP protocol
      * @param name Name of the AES encryption algorithm
      * @param length Length of the encryption key
      */
-    async setSettings(host: string, port: number, protocol: string, name: string, length: number) {
+    async setSettings(storageProvider: number, host: string, port: number, protocol: string, name: string, length: number) {
```

Comment:
I would change `number` to the `StorageProvider` enum so it's easier for users of the library to know what values are supported. (0, 1, 2, 3)
```suggestion
    async setSettings(storageProvider: StorageProvider, host: string, port: number, protocol: string, name: string, length: number) {
```